pkg/dsl/definition/package.go

```diff
@@ -59,6 +62,25 @@ type VersionKind struct {
 	Kind           string
 }
 
+// ParseErrType represents the type of CUEParseError
+type ParseErrType string
+
+// CUEParseError describes an error when CUE parse error
+type CUEParseError struct {
+	err     error
+	errType ParseErrType
+}
+
+// Error implements the Error interface.
+func (cueErr CUEParseError) Error() string {
+	return fmt.Sprintf("%s: %s", cueErr.errType, cueErr.err.Error())
+}
+
+// IsCUEParseErr returns true if the specified error is CUEParseError type.
+func IsCUEParseErr(err error) bool {
+	return errors.As(err, &CUEParseError{})
+}
+
 // NewPackageDiscover will create a PackageDiscover client with the K8s config file.
 func NewPackageDiscover(config *rest.Config) (*PackageDiscover, error) {
```

Comment:
better to have a UT to make sure it works

Reply:
yes, ping @yangsoon

Reply:
got it i will add UT